File: src/main/java/me/centralhardware/znatoki/telegram/statistic/telegram/CommandHandler/ResetCommand.java
package me.centralhardware.znatoki.telegram.statistic.telegram.CommandHandler;

import lombok.RequiredArgsConstructor;
import me.centralhardware.znatoki.telegram.statistic.telegram.fsm.Storage;
import org.springframework.stereotype.Component;
import org.telegram.telegrambots.meta.api.objects.Message;

@Component
@RequiredArgsConstructor
public class ResetCommand extends CommandHandler {

    private final Storage storage;

    @Override
    public void handle (Message message) {
        storage.remove(message.getChatId());
        sender.sendText("Состояние сброшено", message.getFrom());
    }

    @Override
    public boolean isAcceptable (String data) {
        return data.equalsIgnoreCase("/reset");
    }
}

File: src/main/kotlin/me/centralhardware/znatoki/telegram/statistic/telegram/commandHandler/ResetCommand.kt
package me.centralhardware.znatoki.telegram.statistic.telegram.commandHandler

import dev.inmo.tgbotapi.extensions.api.send.sendMessage
import dev.inmo.tgbotapi.extensions.behaviour_builder.BehaviourContext
import dev.inmo.tgbotapi.extensions.behaviour_builder.triggers_handling.onCommand
import dev.inmo.tgbotapi.types.buttons.ReplyKeyboardRemove
import me.centralhardware.znatoki.telegram.statistic.extensions.userId
import me.centralhardware.znatoki.telegram.statistic.telegram.fsm.Storage

suspend fun BehaviourContext.resetCommand() = onCommand("reset") {
    Storage.remove(it.userId())
    sendMessage(it.chat, "Состояние сброшено", replyMarkup = ReplyKeyboardRemove())
}

